gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -73,17 +116,104 @@ public void close() {
       if (instance == null) {
         throw new IllegalStateException("GcpObservability already closed!");
       }
+      unRegisterStackDriverExporter();
       LoggingChannelProvider.shutdown();
       LoggingServerProvider.shutdown();
       sink.close();
       instance = null;
     }
   }
 
-  private GcpObservability(Sink sink,
+  private void setProducer(
+      InternalLoggingChannelInterceptor.Factory channelInterceptorFactory,
+      InternalLoggingServerInterceptor.Factory serverInterceptorFactory) {
+    if (config.isEnableCloudLogging()) {
+      clientInterceptors.add(channelInterceptorFactory.create());
+      serverInterceptors.add(serverInterceptorFactory.create());
+    }
+    if (config.isEnableCloudMonitoring()) {
+      clientInterceptors.add(
+          InternalCensusStatsAccessor.getClientInterceptor(true, true, true, true));
+      tracerFactories.add(
+          InternalCensusStatsAccessor.getServerStreamTracerFactory(true, true, true));
+    }
+    if (config.isEnableCloudTracing()) {
+      clientInterceptors.add(InternalCensusTracingAccessor.getClientInterceptor());
+      tracerFactories.add(InternalCensusTracingAccessor.getServerStreamTracerFactory());
+    }
+
+    if (!clientInterceptors.isEmpty()
+        || !serverInterceptors.isEmpty()
+        || !tracerFactories.isEmpty()) {
+      InternalGlobalInterceptors.setInterceptorsTracers(
+          clientInterceptors, serverInterceptors, tracerFactories);
+    }
+  }
+
+  private void registerStackDriverExporter(String projectId) throws IOException {
+    if (config.isEnableCloudMonitoring()) {
+      RpcViews.registerAllGrpcViews();
+      StackdriverStatsConfiguration.Builder statsConfigurationBuilder =
+          StackdriverStatsConfiguration.builder();
+      if (!Strings.isNullOrEmpty(projectId)) {
+        statsConfigurationBuilder.setProjectId(projectId);
+      }
+      try {
+        StackdriverStatsExporter.createAndRegister(statsConfigurationBuilder.build());
```

Comment:
Using the Static testing class loader we should be able to test this too, right?

Reply:
The reason why we are not able to test `Stackdriver*Exporter` in unit tests is because, `Stackdriver*Exporter` api looks for credentials which is set for Google Cloud environment and fails when run on kokoro.

Reply:
okay, got it